src/Twig.php

```diff
@@ -55,10 +54,7 @@ class Twig implements \ArrayAccess
      */
     public function __construct($path, $settings = [])
     {
-        $this->loader = is_string($path)
-            ? new \Twig_Loader_Filesystem($path)
-            : $this->addPaths($path);
-
+        $this->loader = $this->createLoader(is_string($path) ? [$path] : $path);
```

Comment:
This have more sense to move the checking inside the createLoader method and not in the construct i think.